src/functions-and-arrays.js

```diff
@@ -26,13 +54,36 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(avgNum) {
+  if (avgNum.length == 0){
+    return null;
+  } else {
+    let counter = 0;
+    for (let i=0; i<avgNum.length; i++){
+      return (counter += avgNum[i]) / avgNum.length;
+    }
+    }
+    
+}
+//should return the average even with negative values
+//should return the average of the array
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(word) {
+  if (word.length == 0){
+    return null;
+  } else {
+    let counter = 0;
+    for (let i=0; i<word.length; i++){
+      return (counter += word[i].length) / word.length;
+    }
```

Comment:
- Similar issue here. You want to update counter inside `for loop`, but not return it
- At the moment you are taking each element of array and dividing by the length. However  you'd want to firstly get total sum and only then divide by the length of array